grapher/graph/dataloader_test.go

```diff
@@ -0,0 +1,51 @@
+package graph
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDataloader(t *testing.T) {
+	batchCalls := 0
+	loader := NewDataLoader(
+		newTestBatchFn(&batchCalls),
+		time.Second,
+		time.Millisecond,
+	)
+
+	result, err := loader.Load(context.Background(), "test")
+	assert.Nil(t, err)
+	assert.Equal(t, "test", result)
+
+	result, err = loader.Load(context.Background(), "test")
+	assert.Nil(t, err)
+	assert.Equal(t, "test", result)
+
+	// Cache did not expire yet so the second Load result came from the cache
+	assert.Equal(t, 1, batchCalls)
+
+	time.Sleep(time.Second)
```

Comment:
Not a fan, the ticker needs to be mocked but went with this as the PR was getting too big already.